File: plane_mcp/formatting.py
"""Compact single-line formatters for MCP list responses."""

from plane_mcp import uid


def project(p) -> str:
    """'IDENTIFIER · Name [short_id]'"""
    short = uid.encode(p.id) or "?"
    return f"{p.identifier} · {p.name} [{short}]"


def work_item(item) -> str:
    """'#seq · Name (priority) [short_id]'"""
    short = uid.encode(item.id) or "?"
    priority = _enum_str(item.priority)
    meta = f"({priority})" if priority and priority != "none" else ""
    parts = [f"#{item.sequence_id}", "·", item.name]
    if meta:
        parts.append(meta)
    parts.append(f"[{short}]")
    return " ".join(parts)


def state(s) -> str:
    """'Name (group) [short_id]' — marks default with *"""
    short = uid.encode(s.id) or "?"
    group = _enum_str(s.group) or ""
    default = " *" if s.default else ""
    return f"{s.name} ({group}){default} [{short}]"


def _enum_str(v) -> str | None:
    if v is None:
        return None
    return v.value if hasattr(v, "value") else str(v)

File: formatting/formatting.go
// Package formatting provides compact single-line formatters for MCP list responses.
package formatting

import (
	"fmt"
	"strings"

	"github.com/CaliLuke/plane-mcp-server/uid"
	"golang.org/x/net/html"
)

// Project formats: "IDENTIFIER · Name [short_id]"
func Project(id, name, identifier string) string {
	short := uid.Encode(id)
	if short == "" {
		short = "?"
	}
	return fmt.Sprintf("%s · %s [%s]", identifier, name, short)
}

// WorkItem formats: "PROJ-seq (state) · Name [short_id]"
func WorkItem(id string, sequenceID int, name, projectIdentifier, stateName string) string {
	short := uid.Encode(id)
	if short == "" {
		short = "?"
	}
	var seq string
	if projectIdentifier != "" {
		seq = fmt.Sprintf("%s-%d", projectIdentifier, sequenceID)
	} else {
		seq = fmt.Sprintf("#%d", sequenceID)
	}
	statePart := ""
	if stateName != "" {
		statePart = fmt.Sprintf(" (%s)", stateName)
	}
	return fmt.Sprintf("%s%s · %s [%s]", seq, statePart, name, short)
}

// State formats: "Name (group) [short_id]" — marks default with *
func State(id, name, group string, isDefault bool) string {
	short := uid.Encode(id)
	if short == "" {
		short = "?"
	}
	def := ""
	if isDefault {
		def = " *"
	}
	return fmt.Sprintf("%s (%s)%s [%s]", name, group, def, short)
}

// StripHTML removes HTML tags and returns plain text, or empty string if empty.
func StripHTML(s string) string {
	if s == "" {
		return ""
	}
	tokenizer := html.NewTokenizer(strings.NewReader(s))
	var parts []string
	for {
		tt := tokenizer.Next()
		if tt == html.ErrorToken {
			break
		}
		if tt == html.TextToken {
			text := strings.TrimSpace(tokenizer.Token().Data)
			if text != "" {
				parts = append(parts, text)
			}
		}
	}
	return strings.Join(parts, " ")
}
